frontend/filtering.cpp

```diff
@@ -0,0 +1,50 @@
+#include "mainwindow.h"
+
+// Hide column on view menu change
+void MainWindow::columnHider(int state)
+{
+    QCheckBox *action = qobject_cast<QCheckBox *>(sender());
```

Comment:
Good habit to get into - using `auto`:
```suggestion
    auto *action = qobject_cast<QCheckBox *>(sender());
```